src/utils/utils.rs

```diff
@@ -651,7 +654,21 @@ pub fn toolchain_sort<T: AsRef<str>>(v: &mut Vec<T>) {
 }
 
 fn rename(name: &'static str, src: &Path, dest: &Path) -> Result<()> {
-    fs::rename(src, dest).chain_err(|| ErrorKind::RenamingFile {
+    // https://github.com/rust-lang/rustup.rs/issues/1870
+    // 21 fib steps from 1 sums to ~28 seconds, hopefully more than enough
+    // for our previous poor performance that avoided the race condition with
+    // McAfee and Norton.
+    retry(
+        Fibonacci::from_millis(1).map(jitter).take(21),
+        || match fs::rename(src, dest) {
+            Ok(v) => OperationResult::Ok(v),
+            Err(e) => match e.kind() {
+                io::ErrorKind::PermissionDenied => OperationResult::Retry(e),
```

Comment:
I suppose, on a Linux system (or mac) where the `EPERM` was actually more to do with someone `chown()ing` their stuff oddly, this will end up consuming 28s before failing -- should we fail faster by making this somehow conditional on Windows?